docs/design/2020-08-04-global-index.md

```diff
@@ -0,0 +1,191 @@
+//		|  Padding:       Ensure length of value always >= 10. (or >= 11 if UntouchedFlag exists.)
+//		|  IntHandle:     Only exists when table use int handles and index is unique.
+//		|  UntouchedFlag: Only exists when index is untouched.
+//		|
+//		|  Layout of Options:
+//		|
+//		|     Segment:             Common Handle                 |     Global Index      | New Collation
+//		|     Layout:  CHandle Flag | CHandle Len | CHandle      | PidFlag | PartitionID | restoreData
+//		|     Length:     1         | 2           | len(CHandle) |    1    |    8        | len(restoreData)
+//		|
+//		|     Common Handle Segment: Exists when unique index used common handles.
+//		|     Global Index Segment:  Exists when index is global.
+//		|     New Collation Segment: Exists when new collation is used and index contains non-binary string.
+//		|
+//		+--Old Encoding (without restore data, integer handle, local)
+//
+//		   Layout: [Handle] | [UntouchedFlag]
+//		   Length:   8      |     1
+//
+//		   Handle:        Only exists in unique index.
+//		   UntouchedFlag: Only exists when index is untouched.
+//
+//		   If neither Handle nor UntouchedFlag exists, value will be one single byte '0' (i.e. []byte{'0'}).
+//		   Length of value <= 9, use to distinguish from the new encoding.
+//
+```
+
+ 
+
+### DDL operations
+
+#### Add/Drop Index
+
+All index entries in add by `index.Create`. We modify it to support new encoding, and add `IndexInfo.Global` field to notify workers add global index entries.
+
+Same goes for dropping index.
+
+#### Drop Partition
+
+In global index, all index entries in a partition is not continuous. It is impossible to delete all index entries of a partition in a single  range deletion. So, we add a reorg state (just like add index), which scan records in the partition and remove relative index entries. 
+
+#### Exchange Partition
+
+Also need a reorg state. It is not supported for the time being because of complexity.
+
+#### Drop Table
+
+Add extra one range deletion to clean global indexes.
+
+### Read from global index
+
+In TiDB, operators in partitioned table will be translated to union in the logical plan phase. But a in-progress project plan to translate operators to new partitioned table executors (see issue: [#18016](https://github.com/pingcap/tidb/issues/18016)). we will follow this project and add global index support in new executors.
+
+## Rationale
+
+[Global index in Oracle](https://docs.oracle.com/database/121/VLDBG/GUID-D1E775A0-669B-4E51-8D40-858847B64BEF.htm#VLDBG1256)
+[Global index in OceanBase](https://zhuanlan.zhihu.com/p/48745358)
+
+## Compatibility
+
+MySQL does not support global index, which means this feature may cause some compatibility issues. We add an option  `enable_global_index` in `config.Config`  to control it. The default value of this option is `false`, so TiDB will keep consistent with MySQL, unless user open global index feature manually.
```

Comment:
```suggestion
MySQL does not support global index, which means this feature may cause some compatibility issues. We add an option  `enable_global_index` in `config.Config`  to control it. The default value of this option is `false`, so TiDB will keep consistent with MySQL, unless the user open global index feature manually.
```

Reply:
Maybe it's better to make the config as a global session variable, since it's harder to control which TiDB enable this feature and which do not.